lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,183 @@
+                datapoint.filters[filter] = datapoint.raw.get("Filters").get(filter, "")
+        datapoint.ground_truth_source_documents = add_search_context_to_datapoint(datapoint)
+        datapoint.question = template_Q.format(datapoint.raw["Question"], datapoint.filters.get(filter_Q))
+    return datapoint
+
+
+class Assessment(ABC):
+    @abstractmethod
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        pass
+
+    def __call__(self, context: Context, index: str, **kwargs):
+        return self.run_evaluation(context, index, **kwargs)
+
+
+class QualityAssessment(Assessment):
+    """
+    This Class run Question Answering benchmarking on each datapoint and
+    measure performance
+
+    Attributes:
+        gt_path: Path to ground truth, supports JSON format
+        rag_config: Configration for RAG
+        os_client_args: Configration for connecting to opensearch
+        custom_question_augmentation: Custom String for Augmenting Question
+        question_augmentation_filter: Filters values to be use in custom Question Augmentation
+        metrics: Metrics to test score on, eg document_retrieval_metrics
+        custom_question_augmentation: A string to format questions
+        question_augmentation_filter: A Filter string to be used in formating question
+    """
+
+    def __init__(
+        self,
+        gt_path: str,
+        rag_config: Dict[str, Any],
+        os_client_args: Dict[str, Any],
+        metrics: List[EvaluationMetric] = [document_retrieval_metrics],
+        custom_question_augmentation: str = "{}",
+        question_augmentation_filter: str = "",
+    ):
+        self.user = os.environ.get("USER", os.environ.get("USERNAME"))
+        self.gt_path = gt_path
+        self.rag_config = rag_config
+        self.os_client_args = os_client_args
+        self.metrics = metrics
+        self.custom_question_augmentation = custom_question_augmentation
+        self.question_augmentation_filter = question_augmentation_filter
+
+    @staticmethod
+    def create_evaluation_datapoint(
+        json_dict: Dict, custom_question_augmentation: str = "{}", question_augmentation_filter: str = ""
+    ) -> List:
+        """
+        This method creates evaluation datapoint from JSON file
+        """
+        result = []
+        assert json_dict is not None
+        assert isinstance(json_dict, dict)
+        for datapoint in json_dict["data"][:]:
+            document = EvaluationDataPoint()
+            datapoint["custom_question_augmentation"] = custom_question_augmentation
+            datapoint["question_augmentation_filter"] = question_augmentation_filter
+            document.raw = datapoint
+            document.ground_truth_answer = datapoint["Answer"]
+            document.filters = datapoint.get("Filters", {})
+            result += [{"doc": document.serialize()}]
+        return result
+
+    def run_evaluation(self, ctx: Context, index: str, **kwargs) -> tuple[list[Document], Document]:
```

Comment:
Use `typing.Tuple` not `tuple`.

Reply:
Also `typing.List` not `list`

Reply:
Make sense to use this for making code compatible with pre 3.9 python version.